src/index/base.cpp

```diff
@@ -125,6 +139,25 @@ void BaseIndexNotifications::blockConnected(const interfaces::BlockInfo& block)
 
 void BaseIndexNotifications::blockDisconnected(const interfaces::BlockInfo& block)
 {
+    // During initial sync, ignore validation interface notifications, only
+    // process notifications from sync thread.
+    if (!m_index.m_synced && !block.chain_tip) return;
```

Comment:
f56a851e11321ede9c5bcb43cd067c17f9efab39: why the additional `!block.chain_tip` condition? I'd assume if we happen to experience a reorg during initial sync and disconnect the tip, `block.chain_tip=true` and we don't return, but we wouldn't want to process this.
Possibly related: If, at this commit, I am in the middle of syncing an index and at the same time call `invalidateblock` for the tip, my node coredumps with "`leveldb/db/db_iter.cc:65: virtual leveldb::Slice leveldb::{anonymous}::DBIter::key() const: Assertion `valid_' failed.`"

Reply:
> Interesting, it does sound related. I would be curious to know if is fixed now. And maybe I could make this into a test. I'm curious if you made changes to somehow pause the sync to emulate a reorg happening at the same time?

Yes, I don't encounter this anymore now. What I did was manual testing on regtest (just adding a `std::this_thread::sleep_for` to `ThreadSync`, giving me the possibility to time my `invalidateblock`) - I don't know if it is possible to somehow make an automated test for this that is not racy.